tutorials/s4sdk-resilience/s4sdk-resilience.md

```diff
@@ -37,100 +37,132 @@ In order to avoid such scenarios, we need to equip applications with the ability
+import com.sap.cloud.sdk.odatav2.connectivity.ODataException;
+
+import com.sap.cloud.sdk.s4hana.connectivity.DefaultErpHttpDestination;
+import com.sap.cloud.sdk.s4hana.connectivity.ErpHttpDestination;
+import com.sap.cloud.sdk.s4hana.datamodel.odata.namespaces.businesspartner.AddressEmailAddress;
 import com.sap.cloud.sdk.s4hana.datamodel.odata.namespaces.businesspartner.BusinessPartner;
+import com.sap.cloud.sdk.s4hana.datamodel.odata.namespaces.businesspartner.BusinessPartnerAddress;
+import com.sap.cloud.sdk.s4hana.datamodel.odata.services.BusinessPartnerService;
 import com.sap.cloud.sdk.s4hana.datamodel.odata.services.DefaultBusinessPartnerService;
 
-public class GetBusinessPartnersCommand extends ErpCommand<List<BusinessPartner>> {
-    private static final Logger logger = CloudLoggerFactory.getLogger(GetBusinessPartnersCommand.class);
+public class GetBusinessPartnersCommand {
 
     private static final String CATEGORY_PERSON = "1";
+    private final ErpHttpDestination destination;
+
+    private final BusinessPartnerService businessPartnerService;
+    private final ResilienceConfiguration myResilienceConfig;
 
-    protected GetBusinessPartnersCommand() {
-        super(GetBusinessPartnersCommand.class);
+    public GetBusinessPartnersCommand(String destinationName) {
+        this(destinationName, new DefaultBusinessPartnerService());
     }
 
-    @Override
-    protected List<BusinessPartner> run()
-            throws Exception {
-        final List<BusinessPartner> businessPartners =
-                new DefaultBusinessPartnerService()
-                        .getAllBusinessPartner()
-                        .select(BusinessPartner.BUSINESS_PARTNER,
-                                BusinessPartner.LAST_NAME,
-                                BusinessPartner.FIRST_NAME,
-                                BusinessPartner.IS_MALE,
-                                BusinessPartner.IS_FEMALE,
-                                BusinessPartner.CREATION_DATE)
-                        .filter(BusinessPartner.BUSINESS_PARTNER_CATEGORY.eq(CATEGORY_PERSON))
-                        .orderBy(BusinessPartner.LAST_NAME, Order.ASC)
-                        .top(10)
-                        .execute();
-
-        return businessPartners;
+    public GetBusinessPartnersCommand(String destinationName, BusinessPartnerService service) {
+        destination = DestinationAccessor.getDestination(destinationName).asHttp().decorate(DefaultErpHttpDestination::new);
+        businessPartnerService = service;
+
+        myResilienceConfig = ResilienceConfiguration.of(BusinessPartnerService.class);
     }
 
-    @Override
-    protected List<BusinessPartner> getFallback() {
-        logger.warn("Fallback called because of exception:", getExecutionException());
-        return Collections.emptyList();
+    public List<BusinessPartner> execute() {
+        return ResilienceDecorator.executeSupplier(this::run, myResilienceConfig, e -> Collections.emptyList());
```

Comment:
Please add a fallback that also logs the error somewhere, otherwise debugging issues will be difficult.

Reply:
Updated